servicetalk-dns-discovery-netty/src/test/java/io/servicetalk/dns/discovery/netty/DefaultDnsClientTest.java

```diff
@@ -565,23 +592,25 @@ void unknownHostDiscover() throws Exception {
     }
 
     @Test
-    void singleADiscover() throws Exception {
+    void unknownHostAfterKnownDiscover() throws Exception {
```

Comment:
Consider adding another test with `io.servicetalk.dns.discovery.nxdomain.invalidation=false`. You can make it a pkg-private builder option with default set to the system property value, then you can make changes for each test.